src/community/script/core/src/main/java/org/geoserver/script/rest/service/ScriptService.java

```diff
@@ -0,0 +1,214 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.List;
+
+import static java.lang.String.format;
+
+@Service
+public class ScriptService {
+
+    @Autowired
+    ScriptManager scriptMgr;
+
+    String path;
+
+    public List<Script> getScriptList(HttpServletRequest request) {
+        final String type = (String) request.getAttribute("type");
+        this.path = getPath(request);
+        this.path = stripExtension(this.path);
+
+        org.geoserver.platform.resource.Resource dir = scriptMgr.script(path);
+
+        List<Script> scripts = Lists.newArrayList();
+        if (dir != null) {
+            Filter<Resource> filter =
+                    type != null ? new Filter<org.geoserver.platform.resource.Resource>() {
+                        @Override
+                        public boolean accept(org.geoserver.platform.resource.Resource pathname) {
+                            return type.equalsIgnoreCase(FilenameUtils.getExtension(pathname.name()));
+                        }
+                    } : new Filter<org.geoserver.platform.resource.Resource>() {
+                        @Override
+                        public boolean accept(org.geoserver.platform.resource.Resource pathname) {
+                            return true;
+                        }
+                    };
+            for (org.geoserver.platform.resource.Resource f : Resources.list(dir, filter)) {
+                if (path.equals("apps")) {
+                    org.geoserver.platform.resource.Resource mainScript = scriptMgr.findAppMainScript(f);
+                    if (mainScript != null) {
+                        String name = mainScript.path().substring(
+                                f.parent().path().length() + 1).replace("\\", "/");
+                        scripts.add(new Script(name));
+                    }
+                } else if (path.equals("wps")) {
+                    if (f.getType() == Type.DIRECTORY) {
+                        String namespace = f.name();
+                        List<org.geoserver.platform.resource.Resource> files = f.list();
+                        for(org.geoserver.platform.resource.Resource file: files) {
+                            String name = namespace + ":" + file.name();
+                            scripts.add(new Script(name));
+                        }
+                    } else {
+                        String name = f.name();
+                        scripts.add(new Script(name));
+                    }
+                } else {
+                    String name = f.name();
+                    scripts.add(new Script(name));
+                }
+            }
+        } else {
+            // return empty array, perhaps we should return a 404?
+            // throw new RestletException(format("Could not find script dir %s",
+            // path),
+            // Status.CLIENT_ERROR_NOT_FOUND);
```

Comment:
Why is this commented out? Either fix and uncomment, or remove the else block.

Reply:
fixed.

Reply:
fixed.